src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -968,6 +970,9 @@ export class GuidedTourService {
      * @return guided tour step with defined orientation
      */
     private setTourOrientation(step: TourStep): TourStep {
+        if (!step) {
+            return step;
+        }
```

Comment:
```suggestion
    private setTourOrientation(step: TourStep): TourStep? {
        if (!step) {
            return undefined;
        }
```


Also, I just noticed that this class has quite some instances where `TourStep | undefined` could be simplified as `TourStep?`.